airflow/providers/google/cloud/example_dags/example_cloud_build.yaml

```diff
@@ -0,0 +1,20 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+steps:
```

Comment:
```
Traceback (most recent call last):
  File "/usr/local/bin/airflow", line 11, in <module>
    load_entry_point('apache-airflow', 'console_scripts', 'airflow')()
  File "/opt/airflow/airflow/__main__.py", line 40, in main
    args.func(args)
  File "/opt/airflow/airflow/cli/cli_parser.py", line 52, in command
    return func(*args, **kwargs)
  File "/opt/airflow/airflow/utils/cli.py", line 84, in wrapper
    return f(*args, **kwargs)
  File "/opt/airflow/airflow/cli/commands/task_command.py", line 341, in task_test
    ti.run(ignore_task_deps=True, ignore_ti_state=True, test_mode=True)
  File "/opt/airflow/airflow/utils/session.py", line 61, in wrapper
    return func(*args, **kwargs)
  File "/opt/airflow/airflow/models/taskinstance.py", line 1143, in run
    session=session)
  File "/opt/airflow/airflow/utils/session.py", line 57, in wrapper
    return func(*args, **kwargs)
  File "/opt/airflow/airflow/sentry.py", line 140, in wrapper
    return func(task_instance, *args, session=session, **kwargs)
  File "/opt/airflow/airflow/models/taskinstance.py", line 1024, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/airflow/providers/google/cloud/operators/cloud_build.py", line 217, in execute
    body = BuildProcessor(body=self.body).process_body()
  File "/opt/airflow/airflow/providers/google/cloud/operators/cloud_build.py", line 99, in process_body
    self._verify_source()
  File "/opt/airflow/airflow/providers/google/cloud/operators/cloud_build.py", line 52, in _verify_source
    is_storage = "storageSource" in self.body["source"]
KeyError: 'source'
```
I'm afraid this file won't work. Can you make this optional key?
Can you make this

Reply:
providing a source is optional for a build config. Should we always check and fail if the body misses a source?

Reply:
We should allow the build object to be passed without sources.